control-plane/connect-inject/controllers/endpointsv2/endpoints_controller_test.go

```diff
@@ -54,9 +54,11 @@ type reconcileCase struct {
 	targetConsulNs        string
 	targetConsulPartition string
 	expErr                string
+	caseFn                func(*testing.T, *reconcileCase, *Controller, pbresource.ResourceServiceClient)
 }
 
 // TODO(NET-5716): Allow/deny namespaces for reconcile tests
+// TODO(NET-5932): Add tests for consistently sorting repeated output fields (getConsulService, getServicePorts)
```

Comment:
I added this to improve guards around determinism. Test coverage should be sufficient as-is, so I wanted to avoid blocking review further given time constraints and treat this as an ideal follow-up if there's time.